### ISSUE TITLE
Make pluralSpecs non-global

### ISSUE THREAD

**Issue by bep** [nicksnyder/go-i18n#82]
This replaces the rejected #81

While @nicksnyder is right about thread safety being the client's responsibility, the global `pluralSpecs` makes it harder than it should. 

And then having `RegisterPluralSpec` mutate that state.

In Hugo we run most test with `t.Parallel()`. And while this may report some data races we may not realistically see in real use scenarios, it is well worth it.

In this case, it reports a data race in `pluralSpecs` when using the `RegisterPluralSpec`. We can probably fix that by adding some global locks. 

But these globals also has another unfortunate side-effect:

We cannot run multiple `go-i18n` configurations side-by-side. And this has nothing to do with concurrency. 

So, it would be much better if you could provide a constructor func that takes the options needed (plural specs etc.)

**Comment by nicksnyder** [comment]
I agree that global state is not ideal and regret having a global default bundle as well.

**Comment by nicksnyder** [comment]
V2 contains a proposal to solve this. Please review https://github.com/nicksnyder/go-i18n/pull/92

**Comment by nicksnyder** [comment]
I just tagged [2.0.0.beta.1](https://github.com/nicksnyder/go-i18n/releases/tag/v2.0.0.beta.1). Please start using it and report any issues that you have.
